safe/gui/tools/people_in_buildings_dialog.py

```diff
@@ -0,0 +1,380 @@
+        """Shorthand for clearing and loading options (this happens a lot)
+        """
+        combobox.clear()
+        combobox.addItems(options)
+
+    @staticmethod
+    def _add_attribute_to_layer(layer, attribute_name, data_type):
+        """Add a new attribute to an existing layer.
+
+        :param layer: The layer to be altered.
+        :type layer: VectorLayer
+
+        :param attribute_name: The name of the new attribute.
+        :type attribute_name: basestring
+
+        :param data_type: The type of the new attribute.
+        :type data_type: QVariant.Double, QVariant.Int
+
+        :returns: The success state of the operation.
+        :rtype: bool
+        """
+        provider = layer.dataProvider()
+        # Check if attribute is already there, return "-1" if not
+        ind = provider.fieldNameIndex(attribute_name)
+        if ind != -1:
+            return False
+        field = QgsField(attribute_name, data_type)
+        result = provider.addAttributes([field])
+        return result
+
+    def add_population_attribute(self, layer):
+        """Add people estimate columns/attributes to the output layer.
+
+        :param layer: The layer that the population attribute should be added.
+        :type layer: VectorLayer
+        """
+        attributes = {
+            People_Calculated: QVariant.Double,
+            People_Rounded: QVariant.Int
+        }
+        for attribute_name, data_type in attributes.items():
+            result = self._add_attribute_to_layer(
+                layer, attribute_name, data_type)
+            if result:
+                layer.updateFields()
+
+    @staticmethod
+    def _get_attributes(feature, field_names):
+        """Get the attribute dict based on the feature and the field names.
+
+        :param feature: The feature to be investigated.
+        :type feature:
+
+        :param field_names: The field names list.
+        :type field_names: list
+
+        :return: A lookup of the field values by field name.
+        :rtype: dict
+        """
+        attributes = feature.attributes()
+        attribute_dict = dict(zip(field_names, attributes))
+        return attribute_dict
+
+    @staticmethod
+    def _overlapping_area(feature, layer):
+        """Get the area overlapping area if feature is within layer
+
+        :param feature: A feature
+        :type feature: QgsFeature
+
+        :param layer: A feature
+        :type layer: QgsFeature
+
+        :returns: overlapping area
+        :rtype: float
+        """
+        geometry_feature = feature.geometry()
+        geometry_layer = layer.geometry()
+        centroid = geometry_feature.centroid()
+        if not centroid.within(geometry_layer):
+            return 0
+        return geometry_feature.area()
+
+    @staticmethod
+    def _get_levels(attributes, levels_attribute):
+        """Try to extract the levels from the attributes dict as an int.
+        ..Note: If the levels of this building has not been set we will
```

Comment:
.. note::    (this is rst / sphinx markup and will get rendered as a nice bubble)